sdk/client/client.go

```diff
@@ -217,13 +221,14 @@ func (r *Response) Unmarshal(model interface{}) error {
 		respBody = bytes.TrimPrefix(respBody, []byte("\xef\xbb\xbf"))
 
 		// copy the byte stream across
-		model = respBody
+		reflect.ValueOf(model).Elem().Elem().SetBytes(respBody)
```

Comment:
we should not use  reflect to set the value of `**[]byte` type here, for it will cause panic when `model` is a zero valued pointer. why not use `*model = &respBody` directly?

```go
func SetBytes(model interface{}) error {
	if _, ok := model.(**[]byte); !ok {
		return fmt.Errorf("internal-error: `model` must be **[]byte but got %+v", model)
	}
	respBody := []byte("foo")
	reflect.ValueOf(model).Elem().Elem().SetBytes(respBody)
	return nil
}

func TestSetPtr(t *testing.T) {
	var p *[]byte
	if err := SetBytes(&p); err != nil {  <-- panic: reflect: call of reflect.Value.SetBytes on zero Value
		t.Fatal(err)
	}
	t.Logf("p value %v", *p)
}


////// ===========

func SetBytesV2(model interface{}) error {
	if pptr, ok := model.(**[]byte); !ok {
		return fmt.Errorf("internal-error: `model` must be **[]byte but got %+v", model)
	} else {
		respBody := []byte("foo")
		*pptr = &respBody
	}
	return nil
}

func TestSetPtrV2(t *testing.T) {
	var p *[]byte

	if err := SetBytesV2(&p); err != nil {
		t.Fatal(err)
	}
	t.Logf("p value %v", *p)  <-- set p as "foo" success
}
````